File: Algorithms/557. Reverse Words in a String III/kotlin/solution.reverse.kt
class Solution {
    fun reverseWords(s: String): String {
        var result = ""
        var word = ""

        for (c in s) {
            if (c == ' ') {
                result = result + word + ' '
                word = ""
            } else {
                word = c + word
            }
        }

        result += word
        return result
    }
}

File: Algorithms/557. Reverse Words in a String III/php/solution.reverse.php
<?php
class Solution
{

    /**
     * @param String $s
     * @return String
     */
    function reverseWords($s)
    {
        $result = array();
        $word = '';
        for ($i = 0; $i < strlen($s); $i++) {
            if ($s[$i] == ' ') {
                $result[] = $word;
                $word = '';
            } else {
                $word .= $s[$i];
            }
        }
        $result[] = $word;
        return implode(' ', array_map('strrev', $result));
    }
}
